src/ahocode.py

```diff
@@ -0,0 +1,341 @@
+            except KeyError:
+                next_state = State(self._counter, parent=current_state,
+                                   symbol=char)
+                self._counter += 1
+                current_state.transitions[char] = next_state
+                current_state = next_state
+        current_state.value = value
+        if current_state.success:
+            return False
+        current_state.success = True
+        current_state.matched_keyword = original_keyword
+        return True
+
+    def clear(self):  # real signature unknown; restored from __doc__
+        """
+        clear()
+
+        Remove all keys from the trie. This method invalidates all
+        iterators.
+        """
+        self._zero_state = State(0)
+        self._counter = 1
+        self._finalized = False
+
+    def exists(self, key):  # real signature unknown; restored from __doc__
+        """
+        exists(key) -> boolean
+
+        Return True if the key is present in the trie. Same as using
+        the 'in' keyword.
+        """
+        if not self._finalized:
+            current_state = self._zero_state
+            for char in key:
+                try:
+                    current_state = current_state[char]
+                except KeyError:
+                    return False
+            return True
+        else:
+            self.search_all(key)
+
+    def iter(self, string, start=None, end=None, *args, **kwargs):
+        if not self._finalized:
+            raise ValueError('Automaton has not been finalized. No search allowed. Call make_automaton() first.')
+        zero_state = self._zero_state
+        current_state = zero_state
+        offset = 0
+        if (start and end) is not None:
+            offset = start
+            string = string[start:end + 1]
+        for idx, symbol in enumerate(string):
+            current_state = current_state.transitions.get(
+                symbol, zero_state.transitions.get(symbol, zero_state))
+            state = current_state
+            while state is not zero_state:
+                if state.success:
+                    value = state.value
+                    yield idx + offset, value
+                state = state.longest_strict_suffix
+
+    def get(self, key, default=None):
+        """
+        get(key[, default])
+
+        Return the value associated with the key string.
+
+        Raise a KeyError exception if the key is not in the trie and
+        no default is provided.
+
+        Return the optional default value if provided and the key is
+        not in the trie.
+        """
+        current_state = self._zero_state
+        for char in key:
+            try:
+                current_state = current_state.transitions[char]
+            except KeyError:
+                return default
+        return current_state.value
+
+    def values(self, prefix=None, wildcard=None, how=None):
+        for key, value in list(self.items()):

```

Comment:
I feel like you could get away with not wrapping `self.items()` in a list.
```suggestion
        for key, value in self.items():

```